### ISSUE TITLE
`keybase login` says my key is expired, even after I update it.

### ISSUE THREAD

**Issue by christophebiocca** [keybase/keybase-issues#1955]
This manifests itself for both "Use my Keybase passphrase", "Use GPG - Keybase can use GPG commands to sign the installation." and "Use GPG - Keybase can export your secret key from GPG and save it". I've included `-d` stdout/stderr for all of these:

[login3.err.txt](https://github.com/keybase/keybase-issues/files/85292/login3.err.txt)
[login3.out.txt](https://github.com/keybase/keybase-issues/files/85291/login3.out.txt)
[login4-1.err.txt](https://github.com/keybase/keybase-issues/files/85293/login4-1.err.txt)
[login4-1.out.txt](https://github.com/keybase/keybase-issues/files/85294/login4-1.out.txt)
[login4-2.err.txt](https://github.com/keybase/keybase-issues/files/85295/login4-2.err.txt)
[login4-2.out.txt](https://github.com/keybase/keybase-issues/files/85296/login4-2.out.txt)

Moving the `.gnupg` directory elsewhere and running option 3) again still gives the same error, so I'm pretty sure it has nothing to do with my local key.

I updated the expiration date (to 2018) and reuploaded my public key on the website before doing this, so I'm not sure why it's still using the old expiration date.

**Comment by maxtaco** [comment]
Thanks for the bug report, I'll take a look!

**Comment by maxtaco** [comment]
Repro'd and I understand what's going on.  You have several sigs on your UIDs, some expired, and some not-expired (as you said). However, there's a bug in the Go OpenPGP library that is just consuming the first signature (even though the RFC says to [take the more recent one](https://tools.ietf.org/html/rfc4880#section-5.2.3.3)).

I have a fix in the works, but in the mean time, if you upload a key with only the most recent self-sig on your `@keybase.io` username, we won't get confused, and you'll be unblocked.  Thanks for the feedback and for your patience!

**Comment by christophebiocca** [comment]
I tried deleting the signatures (using `uid 2`, then `delsig`).

First I just deleted the signature on `christophe@keybase.org`.
Then I deleted all outdated signatures on all UIDs

Re-uploaded both times using the web interface.

Neither one worked.

I'm not in a rush, so I'm happy to wait for an official fix, but just wanted to let you know your suggested workaround doesn't seem to work.

**Comment by maxtaco** [comment]
Thanks Christophe.  This might actually be a new bug, since as you said, you did nuke your old signatures.  We'll keep investigating, thanks again.

**Comment by jgjay** [comment]
I think I'm running into the same issue. I've attached stderr from `keybase -d login` to confirm - if there's anything else I can provide to help please let me know. 

[keybase-login.err.txt](https://github.com/keybase/keybase-issues/files/96686/keybase-login.err.txt)

**Comment by maxtaco** [comment]
@jgjay thanks for the bug report, I can't explain what's going on in your case.  It likely has something to do with the many sigchains resets.  We'll take a look

**Comment by maxtaco** [comment]
Which login path did you choose BTW?

**Comment by jgjay** [comment]
The login path I took was `(4) Use GPG` then `(1) Keybase can use GPG commands to sign the installation`. None of the other options worked for me because I lack another device, don't have a paper based key yet and my account already has a public key defined.

Is it worth me opening a separate issue for my case, given that it's not the same as this one? I don't want to confuse or complicate things by discussing two separate issues with similar symptoms in the same thread.

**Comment by maxtaco** [comment]
I think we need to add some more debugging and then ask you to run it
again. We will look into it shortly.

**Comment by maxtaco** [comment]
@jgjay, I see you're on 1.0.8. I think this bug was fixed in 1.0.9. Can you try an upgrade and let me know? Thanks!

**Comment by christophebiocca** [comment]
I'm on 1.0.9 now and was able to log in. Thanks for fixing it!

**Comment by jgjay** [comment]
Upgraded, and logging in worked perfectly. Thanks so much!

**Comment by maxtaco** [comment]
:+1:
